flyable/parse/parser_visitor.py

```diff
@@ -397,7 +397,7 @@ def visit_Name(self, node: Name) -> Any:
         elif isinstance(node.ctx, Store):  # not found so declaring a variable
             found_var = self.__func.get_context().add_var(node.id, self.__assign_type)
             if self.__func.get_parent_func().is_global():
-                var_name = "@global@var" + self.__func.get_parent_func().get_file().get_path()
+                var_name = f"@global@var@{found_var.get_name()}@{self.__func.get_parent_func().get_file().get_path()}"
```

Comment:
We were using the same variable name when creating new variables here so they were being overwritten 
@ALavallee